pkg/client/vault/client.go

```diff
@@ -170,16 +203,84 @@ func (c *Client) CreateVault() (*CreatedVault, error) {
 	}, nil
 }
 
+// CreateAuthorization creates a new authorization.
+// nolint: funlen
+func (c *Client) CreateAuthorization(vaultID, requestingParty string, scope *Scope) (*CreatedAuthorization, error) {
+	info, err := c.getVaultInfo(vaultID)
+	if err != nil {
+		return nil, fmt.Errorf("get vault info: %w", err)
+	}
+
+	kh, err := c.kms.Get(info.KID)
+	if err != nil {
+		return nil, fmt.Errorf("kms get: %w", err)
+	}
+
+	kmsCapability, err := uncompressZCAP(info.Auth.KMS.AuthToken)
+	if err != nil {
+		return nil, fmt.Errorf("kms uncompressZCAP: %w", err)
+	}
+
+	kmsNewCapability, err := zcapld.NewCapability(&zcapld.Signer{
+		SignatureSuite:     ed25519signature2018.New(suite.WithSigner(newSigner(c.crypto, kh))),
+		SuiteType:          ed25519signature2018.SignatureType,
+		VerificationMethod: vaultID,
+	}, zcapld.WithParent(kmsCapability.ID), zcapld.WithInvoker(requestingParty),
+		zcapld.WithAllowedActions("unwrap"),
+		zcapld.WithInvocationTarget(kmsCapability.InvocationTarget.ID, kmsCapability.InvocationTarget.Type),
+		zcapld.WithCapabilityChain(kmsCapability.ID))
+	if err != nil {
+		return nil, fmt.Errorf("kms new capability: %w", err)
+	}
+
+	kmsCompressedCapability, err := compressZCAP(kmsNewCapability)
+	if err != nil {
+		return nil, fmt.Errorf("kms compressZCAP: %w", err)
+	}
+
+	edvCapability, err := uncompressZCAP(info.Auth.EDV.AuthToken)
+	if err != nil {
+		return nil, fmt.Errorf("edv uncompressZCAP: %w", err)
+	}
+
+	edvNewCapability, err := zcapld.NewCapability(&zcapld.Signer{
+		SignatureSuite:     ed25519signature2018.New(suite.WithSigner(newSigner(c.crypto, kh))),
+		SuiteType:          ed25519signature2018.SignatureType,
+		VerificationMethod: vaultID,
+	}, zcapld.WithParent(edvCapability.ID), zcapld.WithInvoker(requestingParty),
+		zcapld.WithAllowedActions(scope.Actions...),
+		zcapld.WithInvocationTarget(edvCapability.InvocationTarget.ID, edvCapability.InvocationTarget.Type),
```

Comment:
Same for the EDV (see https://github.com/trustbloc/edge-service/pull/601/files#r575187335)